File: ios/Framework/Core/Models/Asset.swift
/**
* Copyright (c) 2000-present Liferay, Inc. All rights reserved.
*
* This library is free software; you can redistribute it and/or modify it under
* the terms of the GNU Lesser General Public License as published by the Free
* Software Foundation; either version 2.1 of the License, or (at your option)
* any later version.
*
* This library is distributed in the hope that it will be useful, but WITHOUT
* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
* FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
* details.
*/
import Foundation


@objc public class Asset : NSObject, NSCoding, MimeTypeable {

	public let attributes :[String:AnyObject]

	public let title: String


	//MARK: MimeTypeable

	public var mimeType: String? {
		return attributes["mimeType"]?.description
	}

	override public var description: String {
		return attributes["description"] as! String
	}

	public var classNameId: Int64 {
		return attributes["classNameId"]!.longLongValue
	}

	public var classPK: Int64 {
		return attributes["classPK"]!.longLongValue
	}

	public var groupId: Int64 {
		return attributes["groupId"]!.longLongValue
	}

	public var companyId: Int64 {
		return attributes["companyId"]!.longLongValue
	}

	public var url: String {
		return attributes["url"]!.description!
	}

	public var entryId: Int64 {
		return attributes["entryId"]!.longLongValue!
	}

	public var createDate: NSDate {
		let milliseconds = (attributes["createDate"]! as! NSNumber).doubleValue
		return NSDate(millisecondsSince1970: milliseconds)
	}

	public var modifiedDate: NSDate {
		let milliseconds = (attributes["modifiedDate"]! as! NSNumber).doubleValue
		return NSDate(millisecondsSince1970: milliseconds)
	}

	override public var debugDescription: String {
		return attributes.debugDescription
	}

	//MARK: Initializers

	public init(attributes: [String:AnyObject]) {
		self.attributes = attributes

		let xmlTitle = attributes["title"] as! String
		title = xmlTitle.asLocalized(NSLocale(localeIdentifier: NSLocale.currentLocaleString))

		super.init()
	}

	public required init?(coder aDecoder: NSCoder) {
		self.attributes = aDecoder.decodeObjectForKey("asset-attrs") as? [String:AnyObject] ?? [:]

		let xmlTitle = attributes["title"] as! String
		title = xmlTitle.asLocalized(NSLocale(localeIdentifier: NSLocale.currentLocaleString))

		super.init()
	}


	//MARK: Public methods

	public func encodeWithCoder(aCoder: NSCoder) {
		aCoder.encodeObject(attributes, forKey: "asset-attrs")
	}
	
}

File: xamarin/Framework/LiferayScreens.iOS/Models/Asset.cs
using Foundation;
using ObjCRuntime;
using System;

namespace LiferayScreens
{
    // @interface Asset : NSObject <NSCoding>
    [BaseType(typeof(NSObject))]
    [DisableDefaultCtor]
    interface Asset : INSCoding
    {
        // @property (readonly, copy, nonatomic) NSDictionary<NSString *,id> * _Nonnull attributes;
        [Export("attributes", ArgumentSemantic.Copy)]
        NSDictionary<NSString, NSObject> Attributes { get; }

        // @property (readonly, copy, nonatomic) NSString * _Nonnull title;
        [Export("title")]
        string Title { get; }

        // @property (readonly, copy, nonatomic) NSString * _Nullable mimeType;
        [NullAllowed, Export("mimeType")]
        string MimeType { get; }

        // @property (readonly, copy, nonatomic) NSString * _Nonnull description;
        [Export("description")]
        string Description { get; }

        // @property (readonly, nonatomic) int64_t classNameId;
        [Export("classNameId")]
        long ClassNameId { get; }

        // @property (readonly, nonatomic) int64_t classPK;
        [Export("classPK")]
        long ClassPK { get; }

        // @property (readonly, nonatomic) int64_t groupId;
        [Export("groupId")]
        long GroupId { get; }

        // @property (readonly, nonatomic) int64_t companyId;
        [Export("companyId")]
        long CompanyId { get; }

        // @property (readonly, copy, nonatomic) NSString * _Nonnull url;
        [Export("url")]
        string Url { get; }

        // @property (readonly, nonatomic) int64_t entryId;
        [Export("entryId")]
        long EntryId { get; }

        // @property (readonly, copy, nonatomic) NSDate * _Nonnull createDate;
        [Export("createDate", ArgumentSemantic.Copy)]
        NSDate CreateDate { get; }

        // @property (readonly, copy, nonatomic) NSDate * _Nonnull modifiedDate;
        [Export("modifiedDate", ArgumentSemantic.Copy)]
        NSDate ModifiedDate { get; }

        // @property (readonly, copy, nonatomic) NSString * _Nonnull debugDescription;
        [Export("debugDescription")]
        string DebugDescription { get; }

        // -(instancetype _Nonnull)initWithAttributes:(NSDictionary<NSString *,id> * _Nonnull)attributes __attribute__((objc_designated_initializer));
        [Export("initWithAttributes:")]
        [DesignatedInitializer]
        IntPtr Constructor(NSDictionary<NSString, NSObject> attributes);

        // -(void)encodeWithCoder:(NSCoder * _Nonnull)aCoder;
        [Export("encodeWithCoderAsset:")]
        void EncodeWithCoder(NSCoder aCoder);
    }
}
